### PR TITLE
Fixed product sorting by category

### PR DESCRIPTION
**Pull request by thierrymarianne**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Sort product by category from the back-office |
| Type? | bug fix |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1376 |
| How to test? | Products in front-office should be sorted accordingly |

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 1]
```diff
@@ -622,7 +622,12 @@ public function massEditAction(Request $request, $action)
                         ['product_list_id' => $productIdList, 'product_list_position' => $productPositionList]
                     );
                     // Hooks: managed in ProductUpdater
-                    $productUpdater->sortProductIdList(array_combine($productIdList, $productPositionList), $productProvider->getPersistedFilterParameters());
+                    $persistedFilterParams = $productProvider->getPersistedFilterParameters();
+                    $productList = array_combine($productIdList, $productPositionList);
+                    $productUpdater->sortProductIdList(
+                        $productList,
+                        ['filter_category' => $persistedFilterParams['filter_category']]
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: can you keep the long ``array()`` notation please ? sorry :/

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 2]
```diff
@@ -622,7 +622,12 @@ public function massEditAction(Request $request, $action)
...
+                    );
                     $this->addFlash('success', $translator->trans('Products successfully sorted.', [], 'Admin.Catalog.Notification'));
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: I know this file have some coding style issue, should we update it now ?

File: src/PrestaShopBundle/Controller/Admin/ProductController.php (hunk 1 above)
Comment (thierrymarianne) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: sorry, hard to change habits :/

Comment (thierrymarianne) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: I was influenced by the lines above and below, which I'm going to fix as well.

**Comment by vincentbz** [comment]
Tested and ok, thanks !

**Comment by maximebiloe** [comment]
Thanks!

**Pull request merged**